### PR TITLE
Require MPI for Trilinos

### PR DESCRIPTION
**Pull request by masterleinad**
Supersedes https://github.com/dealii/dealii/pull/12343 by requring MPI for Trilinos. The discussion is still going on but so far everyone is in favor of simply assuming MPI is available when using Trilinos.

### CODE REVIEW

File: include/deal.II/lac/read_write_vector.h [hunk 1]
```diff
@@ -697,7 +693,7 @@ namespace LinearAlgebra
     void
     resize_val(const size_type new_allocated_size);
 
-#if defined(DEAL_II_WITH_TRILINOS) && defined(DEAL_II_WITH_MPI)
+#if defined(DEAL_II_WITH_TRILINOS)
```
Comment (tjhei) [inline include/deal.II/lac/read_write_vector.h:696]:
```suggestion
#ifdef DEAL_II_WITH_TRILINOS
```

File: include/deal.II/dofs/dof_accessor.templates.h [hunk 2]
```diff
@@ -705,7 +705,7 @@ namespace internal
 
 
 
-#if defined(DEAL_II_WITH_TRILINOS) && defined(DEAL_II_WITH_MPI)
+#if defined(DEAL_II_WITH_TRILINOS)
```
Comment (tjhei) [inline include/deal.II/dofs/dof_accessor.templates.h:708]:
```suggestion
#ifdef DEAL_II_WITH_TRILINOS
```

File: include/deal.II/lac/read_write_vector.templates.h [hunk 3]
```diff
@@ -299,7 +299,7 @@ namespace LinearAlgebra
 
 
 
-#if defined(DEAL_II_WITH_TRILINOS) && defined(DEAL_II_WITH_MPI)
+#if defined(DEAL_II_WITH_TRILINOS)
```
Comment (tjhei) [inline include/deal.II/lac/read_write_vector.templates.h:302]: and all these places...

File: source/lac/trilinos_vector.cc [hunk 4]
```diff
@@ -343,7 +335,7 @@ namespace TrilinosWrappers
         }
       else
         vector = std::move(actual_vec);
-#  if defined(DEBUG) && defined(DEAL_II_WITH_MPI)
+#  if defined(DEBUG)
```
Comment (tjhei) [inline source/lac/trilinos_vector.cc:338]:
```suggestion
#  ifdef DEBUG
```

**Review by kronbichler — APPROVED** [review]

**Comment by drwells** [comment]
Several people have approved of this change and no one has spoken out against it so lets merge.

**Pull request merged**